--- a/drivers/amlogic/dvb_tv/aml_dvb.c	2016-07-20 16:13:01.000000000 +0200
+++ b/drivers/amlogic/dvb_tv/aml_dvb.c	2016-10-01 23:03:03.164708000 +0200
@@ -506,7 +506,7 @@
 				struct class_attribute *attr, const char *buf,
 				size_t size)
 {
-	enum dmx_source_t src = -1;
+	dmx_source_t src = -1;
 
 	if (!strncmp("ts0", buf, 3))
 		src = DMX_SOURCE_FRONT0;
@@ -573,7 +573,7 @@
 static ssize_t dsc##i##_store_source(struct class *class,  \
 		struct class_attribute *attr, const char *buf, size_t size)\
 {\
-	enum dmx_source_t src = -1, dst = -1;\
+	dmx_source_t src = -1, dst = -1;\
 	\
 	if (!strncmp("dmx0", buf, 4)) {\
 		src = DMX_SOURCE_FRONT0 + 100;\
@@ -675,7 +675,7 @@
 				struct class_attribute *attr, const char *buf,
 				size_t size)
 {
-	enum dmx_source_t src = -1;
+	dmx_source_t src = -1;
 
 	if (!strncmp("ts0", buf, 3))
 		src = DMX_SOURCE_FRONT0;
@@ -747,7 +747,7 @@
 static ssize_t demux##i##_store_source(struct class *class,  \
 		struct class_attribute *attr, const char *buf, size_t size)\
 {\
-	enum dmx_source_t src = -1;\
+	dmx_source_t src = -1;\
 	\
 	if (!strncmp("ts0", buf, 3)) {\
 		src = DMX_SOURCE_FRONT0;\
