--- /dev/null	2016-09-28 17:47:14.234734000 +0200
+++ b/drivers/amlogic/dvb_tv/avl6862.c	2016-10-01 21:59:27.800708000 +0200
@@ -0,0 +1,23 @@
+/*
+* AMLOGIC DVB avl6862 dummy driver.
+*/
+
+#include <linux/version.h>
+#include <linux/kernel.h>
+#include <linux/module.h>
+
+
+static int __init avl6862_init(void)
+{
+	return 0;
+}
+
+static void __exit avl6862_exit(void)
+{
+}
+
+module_init(avl6862_init);
+module_exit(avl6862_exit);
+
+MODULE_DESCRIPTION("amlogic avl6862 driver");
+MODULE_AUTHOR("afl1");
