--- a/include/uapi/linux/dvb/dmx.h	2016-10-01 22:50:46.836708000 +0200
+++ b/include/uapi/linux/dvb/dmx.h	2016-10-01 22:53:14.128708000 +0200
@@ -105,6 +105,7 @@
 #define DMX_ONESHOT         2
 #define DMX_IMMEDIATE_START 4
 #define DMX_KERNEL_CLIENT   0x8000
+#define DMX_USE_SWFILTER    0x100
 };
 
 
@@ -130,7 +131,10 @@
 	DMX_SOURCE_DVR0   = 16,
 	DMX_SOURCE_DVR1,
 	DMX_SOURCE_DVR2,
-	DMX_SOURCE_DVR3
+	DMX_SOURCE_DVR3,
+	DMX_SOURCE_FRONT0_OFFSET = 100,
+	DMX_SOURCE_FRONT1_OFFSET,
+	DMX_SOURCE_FRONT2_OFFSET
 } dmx_source_t;
 
 struct dmx_stc {
