--- a/drivers/amlogic/dvb_tv/Makefile	2016-07-20 16:13:01.000000000 +0200
+++ b/drivers/amlogic/dvb_tv/Makefile	2016-10-01 12:48:17.460708000 +0200
@@ -8,6 +8,9 @@
 
 obj-$(CONFIG_AM_DVB) += aml_fe.o
 
+obj-$(CONFIG_AM_AVL6862) += avl6862.o AVL_Demod_CommonAPI.o AVL_Demod_CommonInternal.o\
+		 AVL_Demod_DVBC.o AVL_Demod_DVBSx.o AVL_Demod_DVBTx.o AVL_Demod_ISDBT.o AVL_Demod_Config.o user_defined_function.o
+obj-$(CONFIG_AM_R848) += r848.o
 #obj-$(CONFIG_AM_CI) += amlci/
 
 obj-$(CONFIG_AM_DTVDEMOD) += amldemod/
